### ISSUE TITLE
Multi-frame viewer focus styling on button and input

### ISSUE THREAD

**Issue by mcbouslog** [zooniverse/front-end-monorepo#1591]
As identified in #1578. 

The focus styling for the next and previous buttons are being handled by Grommet, whereas we're manually handling the styling for the inputs list, so it's possible to have both have focus styling state:

<img width="136" alt="Screen Shot 2020-03-26 at 11 50 47 AM" src="https://user-images.githubusercontent.com/5016731/77673537-4c87e100-6f58-11ea-9190-e4bc849b6780.png">

I got this to happen by mixing my interaction between the mouse and keyboard. I tabbed over into the list and used the arrow keys, then used the previous button, then tabbed back and used the arrow keys. Then both the button and the and the input have the focus styling. There's been long discussion about how the focus styling applies to buttons and the Grommet team thinks this is correct. The focus state remains on the button after you click on it. I think the issue here is the mix between using Grommet and our own components and styling.

**Comment by eatyourgreens** [comment]
I'm unable to reproduce this on https://frontend.preview.zooniverse.org/projects/bostonpubliclibrary/anti-slavery-manuscripts/classify

Here's what I did (in Chrome 80):
- Tab to the image thumbnails.
- Change the selected image with the cursor keys.
- Clicked Previous.
- Tabbed back into the radio button group and selected a new radio button group with the Tab key.

The selected thumbnail has a focus ring and the Previous button looks fine.
![Screenshot of the image viewer showing one thumbnail as having focus.](https://user-images.githubusercontent.com/59547/78244429-81a6ad00-74dd-11ea-8268-b9162fd5f407.png)

**Comment by eatyourgreens** [comment]
Update to my previous comment: I can reproduce this in Chrome 80 on MacOS Catalina, but not on Mojave. It's weirder than the original description. If I click the Previous button, then tab to the Next button _while still hovering the cursor over the Previous button_ then focus rings appear around both Grommet buttons. Only the Next button is actually focused.

![Screenshot of the multi-frame subject viewer with focus rings around both the Previous button and Next button.](https://user-images.githubusercontent.com/59547/78643292-f4cc6c80-78ab-11ea-8332-9a514c5c3130.png)

**Comment by eatyourgreens** [comment]
I've figured out the bug here (and also how to reproduce it in any browser/OS.) The underlying cause is that disabled buttons don't fire events, including `focus` and `blur`.

To reproduce: focus the Previous button, but also disable it, by using it to select the first thumbnail. Now tab or click to shift focus to any other focusable element. Focus (and `document.activeElement`) changes to the new element. `blur` is _not_ fired for the disabled button, so `setFocus(false)` is not run here in the button code:
https://github.com/grommet/grommet/blob/82ae3e014bbbfed73dc5d53cb7ef75af0e00d113/src/js/components/Button/Button.js#L130-L133

TLDR: it's a Grommet bug caused by a combination of the button being disabled and Grommet tracking focus in component state. The fix might be to set focus state to false for disabled buttons, since they cannot be focused.

**Comment by eatyourgreens** [comment]
Here's a screenshot from Chrome 80 on Mojave, where I used the technique mentioned above to add focus rings to the Previous button, the Next button and one of the thumbnails, just by clicking them in a particular order (no tabbing.)

![Screenshot of the thumbnail selector with focus rings around the Previous button, the Next button and one of the thumbnails.](https://user-images.githubusercontent.com/59547/78654129-09186580-78bc-11ea-89e1-aa72ea2fc9ae.png)

**Comment by mcbouslog** [comment]
Thank you @eatyourgreens!! Your fix noted "to set focus state to false for disabled buttons, since they cannot be focused" makes sense. This use case that includes a click that also disables it is indeed likely an outlier. Maybe we don't disable the buttons, but style and control similarly, though only if that can be done in an accessible way.

**Comment by srallen** [comment]
Or how about we contribute a fix back to Grommet?

**Comment by mcbouslog** [comment]
I think the focus state to false for disabled buttons makes sense to contribute back to Grommet, regardless of where we land with this multi-frame viewer.

The active element disabled and focus state for the Previous/Next buttons when the first/last item is selected by the button is tough, or for me to think about anyway. On one hand it makes sense a disabled button shouldn't be focusable, but in this frame viewer use case, if I click the Previous button to select the first frame, I think I'd expect focus to stay on the button? I'm not really sure on either point though tbh.

**Comment by srallen** [comment]
First step would be to open an issue with them and see what they say about it. They're very responsive.

**Comment by srallen** [comment]
The focus styling on disabled state is fixed in https://github.com/grommet/grommet/issues/4393, so we just need to bump our grommet version.

**Comment by mcbouslog** [comment]
Closed by #1883
